tests/manage/z_cluster/cluster_expansion/conftest.py

```diff
@@ -0,0 +1,100 @@
+        Args:
+            ocs_nodes (bool): True if new nodes are OCS, False otherwise
+            node_count (int): Number of nodes to be added
+            taint_label (str): Taint label to be added
+
+        """
+
+        new_nodes = []
+        if config.ENV_DATA["platform"].lower() in constants.CLOUD_PLATFORMS:
+            dt = config.ENV_DATA["deployment_type"]
+            if dt == "ipi":
+                machines = machine_utils.get_machinesets()
+                log.info(
+                    f"The worker nodes number before expansion {len(get_worker_nodes())}"
+                )
+                for machine in machines:
+                    new_nodes.append(
+                        add_new_node_and_label_it(machine, mark_for_ocs_label=ocs_nodes)
+                    )
+
+                log.info(
+                    f"The worker nodes number after expansion {len(get_worker_nodes())}"
+                )
+
+            else:
+                log.info(
+                    f"The worker nodes number before expansion {len(get_worker_nodes())}"
+                )
+                if config.ENV_DATA.get("rhel_workers"):
+                    node_type = constants.RHEL_OS
+                else:
+                    node_type = constants.RHCOS
+
+                new_nodes.append(
+                    add_new_node_and_label_upi(
+                        node_type, node_count, mark_for_ocs_label=ocs_nodes
+                    )
+                )
+                log.info(
+                    f"The worker nodes number after expansion {len(get_worker_nodes())}"
+                )
+
+        elif config.ENV_DATA["platform"].lower() == constants.VSPHERE_PLATFORM:
+            # pytest.skip(
+            #     "Skipping add node in Vmware platform due to "
+            #     "https://bugzilla.redhat.com/show_bug.cgi?id=1844521"
+            # )
+            # Issue to remove skip code https://github.com/red-hat-storage/ocs-ci/issues/2403
+            log.info(
+                f"The worker nodes number before expansion {len(get_worker_nodes())}"
+            )
+            if config.ENV_DATA.get("rhel_user"):
+                pytest.skip("Skipping add RHEL node, code unavailable")
+            node_type = constants.RHCOS
+            new_nodes.append(
+                add_new_node_and_label_upi(node_type, num_nodes=node_count)
+            )
+            log.info(
+                f"The worker nodes number after expansion {len(get_worker_nodes())}"
+            )
+
+        def func(list_to_flatten):
+            flat_list = []
+            for li in list_to_flatten:
+                for sublist in li:
+                    flat_list.append(sublist)
+            return flat_list
+
+        func(new_nodes)
+        if taint_label:
+            taint_nodes(
+                nodes=func(new_nodes), taint_label=taint_label
+            ), "Failed to taint nodes"
+        log.info(f"Successfully Tainted nodes {new_nodes} with {taint_label}")
```

Comment:
What about just use nested list comprehension  for that 
```suggestion
      nodes_to_taint = [node for sublist in new_nodes for node in sublist]
        if taint_label:
            taint_nodes(
                nodes=func(nodes_to_taint), taint_label=taint_label
            ), "Failed to taint nodes"
        log.info(f"Successfully Tainted nodes {nodes_to_taint} with {taint_label}")
```
Or lambda function but this func called func is not nice here.

Also not sure how big the list of nodes looks like in this log, can you give me some example?
Not sure out of my head if it will give us whole dictionary of node or some non sence name of object. If so I prefer to print something like [node.name for node in nodes_to_taing] to print just name of nodes here.

Reply:
done